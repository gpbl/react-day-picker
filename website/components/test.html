<!DOCTYPE html>
<html>
<head>
<style>
body {
  display: flex;
  margin: 0;
  padding: 0;
  min-height: 100vh;
  overflow-x: hidden;
}

.sidebar {
  position: sticky;
  top: 0;
  height: 100vh;
  width: 200px;
  background-color: #f5f5f5;
  padding: 20px;
  overflow-y: auto;
}

.main-content {
  flex-grow: 1;
  padding: 20px;
}
</style>
</head>
<body>

<div class="sidebar">
  <h2>Sidebar</h2>
  <p>This is a sticky sidebar.</p>
</div>

<div class="main-content">
  <h2>Main Content</h2>
  <p>Scroll the page.</p>
  <!-- Add enough content here so that the page scrolls -->
</div>

</body>
</html>
