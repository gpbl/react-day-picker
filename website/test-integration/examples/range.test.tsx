import React from 'react';

import { axe } from '@site/test/axe';
import { user } from '@site/test/user';
import { act, render } from '@testing-library/react';
import { addDays } from 'date-fns';

import {
  getAllSelectedDays,
  getDayButton
} from 'react-day-picker/test/selectors';

import Example from '@examples/range';

const pastMonth = new Date(2020, 10, 15);
let container: HTMLElement;
const days = [
  pastMonth,
  addDays(pastMonth, 1),
  addDays(pastMonth, 2),
  addDays(pastMonth, 3),
  addDays(pastMonth, 4)
];

beforeEach(() => (container = render(<Example />).container));

test('should not have AXE violations', async () => {
  expect(await axe(container)).toHaveNoViolations();
});

test('should match the snapshot', () => {
  expect(container).toMatchSnapshot();
});
test.each(days)('%s should be selected', (day) => {
  expect(getDayButton(day)).toHaveAttribute('aria-selected', 'true');
});

describe('when a day in the range is clicked', () => {
  const day = days[2];
  beforeEach(async () => act(() => user.click(getDayButton(day))));
  test.each([days[0], days[1], day])('%s should be selected', (day) => {
    expect(getDayButton(day)).toHaveAttribute('aria-selected', 'true');
  });
  test.each([days[3], days[4]])('%s should not be selected', (day) => {
    expect(getDayButton(day)).not.toHaveAttribute('aria-selected');
  });
  describe('when the day is clicked again', () => {
    const day = days[2];
    beforeEach(async () => act(() => user.click(getDayButton(day))));
    test('only one day should be selected', () => {
      expect(getAllSelectedDays()).toHaveLength(1);
    });
    test('only a day in the range should be selected', () => {
      expect(getDayButton(day)).toHaveAttribute('aria-selected', 'true');
    });

    describe('when a day in the range is clicked again', () => {
      const day = days[2];
      beforeEach(async () => act(() => user.click(getDayButton(day))));
      test('only one day should be selected', () => {
        expect(getAllSelectedDays()).toHaveLength(1);
      });
      test('should match the snapshot', () => {
        expect(container).toMatchSnapshot();
      });
    });
  });
});
