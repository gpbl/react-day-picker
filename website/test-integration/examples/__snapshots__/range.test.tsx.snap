// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div
    class="rdp "
    id="test"
  >
    <div
      class="rdp-months"
    >
      <div
        class="rdp-month rdp-caption_start rdp-caption_end"
      >
        <div
          class="rdp-caption"
        >
          <div
            aria-live="polite"
            class="rdp-caption_label"
            id="test-0"
            role="presentation"
          >
            November 2020
          </div>
          <div
            class="rdp-nav"
          >
            <button
              aria-label="Go to previous month"
              class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_previous"
              name="previous-month"
              type="button"
            >
              <svg
                class="rdp-nav_icon"
                height="16px"
                viewBox="0 0 120 120"
                width="16px"
              >
                <path
                  d="M69.490332,3.34314575 C72.6145263,0.218951416 77.6798462,0.218951416 80.8040405,3.34314575 C83.8617626,6.40086786 83.9268205,11.3179931 80.9992143,14.4548388 L80.8040405,14.6568542 L35.461,60 L80.8040405,105.343146 C83.8617626,108.400868 83.9268205,113.317993 80.9992143,116.454839 L80.8040405,116.656854 C77.7463184,119.714576 72.8291931,119.779634 69.6923475,116.852028 L69.490332,116.656854 L18.490332,65.6568542 C15.4326099,62.5991321 15.367552,57.6820069 18.2951583,54.5451612 L18.490332,54.3431458 L69.490332,3.34314575 Z"
                  fill="currentColor"
                  fill-rule="nonzero"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next month"
              class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_next"
              name="next-month"
              type="button"
            >
              <svg
                class="rdp-nav_icon"
                height="16px"
                viewBox="0 0 120 120"
                width="16px"
              >
                <path
                  d="M49.8040405,3.34314575 C46.6798462,0.218951416 41.6145263,0.218951416 38.490332,3.34314575 C35.4326099,6.40086786 35.367552,11.3179931 38.2951583,14.4548388 L38.490332,14.6568542 L83.8333725,60 L38.490332,105.343146 C35.4326099,108.400868 35.367552,113.317993 38.2951583,116.454839 L38.490332,116.656854 C41.5480541,119.714576 46.4651794,119.779634 49.602025,116.852028 L49.8040405,116.656854 L100.804041,65.6568542 C103.861763,62.5991321 103.926821,57.6820069 100.999214,54.5451612 L100.804041,54.3431458 L49.8040405,3.34314575 Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <table
          aria-labelledby="test-0"
          class="rdp-table"
          id="test-grid-0"
          role="grid"
        >
          <thead
            class="rdp-head"
          >
            <tr
              class="rdp-head_row"
            >
              <th
                aria-label="Sunday"
                class="rdp-head_cell"
                scope="col"
              >
                Su
              </th>
              <th
                aria-label="Monday"
                class="rdp-head_cell"
                scope="col"
              >
                Mo
              </th>
              <th
                aria-label="Tuesday"
                class="rdp-head_cell"
                scope="col"
              >
                Tu
              </th>
              <th
                aria-label="Wednesday"
                class="rdp-head_cell"
                scope="col"
              >
                We
              </th>
              <th
                aria-label="Thursday"
                class="rdp-head_cell"
                scope="col"
              >
                Th
              </th>
              <th
                aria-label="Friday"
                class="rdp-head_cell"
                scope="col"
              >
                Fr
              </th>
              <th
                aria-label="Saturday"
                class="rdp-head_cell"
                scope="col"
              >
                Sa
              </th>
            </tr>
          </thead>
          <tbody
            class="rdp-tbody"
          >
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  1
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  2
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  3
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  4
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  5
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  6
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  7
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  8
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  9
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  10
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  11
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  12
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  13
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  14
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  aria-selected="true"
                  class="rdp-button_reset rdp-button rdp-day rdp-day_selected rdp-day_range_start"
                  name="day"
                  role="gridcell"
                  tabindex="0"
                  type="button"
                >
                  15
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  aria-selected="true"
                  class="rdp-button_reset rdp-button rdp-day rdp-day_selected rdp-day_range_middle"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  16
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  aria-selected="true"
                  class="rdp-button_reset rdp-button rdp-day rdp-day_selected rdp-day_range_middle"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  17
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  aria-selected="true"
                  class="rdp-button_reset rdp-button rdp-day rdp-day_selected rdp-day_range_middle"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  18
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  aria-selected="true"
                  class="rdp-button_reset rdp-button rdp-day rdp-day_selected rdp-day_range_end"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  19
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  20
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  21
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  22
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  23
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  24
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  25
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  26
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  27
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  28
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  29
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  30
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
            </tr>
          </tbody>
          <tfoot
            class="rdp-tfoot"
          >
            <tr>
              <td
                colspan="8"
              >
                <p>
                  November 15th, 2020
                  –
                  November 19th, 2020
                </p>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`when a day in the range is clicked when the day is clicked again when a day in the range is clicked again should match the snapshot 1`] = `
<div>
  <div
    class="rdp "
    id="test"
  >
    <div
      class="rdp-months"
    >
      <div
        class="rdp-month rdp-caption_start rdp-caption_end"
      >
        <div
          class="rdp-caption"
        >
          <div
            aria-live="polite"
            class="rdp-caption_label"
            id="test-0"
            role="presentation"
          >
            November 2020
          </div>
          <div
            class="rdp-nav"
          >
            <button
              aria-label="Go to previous month"
              class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_previous"
              name="previous-month"
              type="button"
            >
              <svg
                class="rdp-nav_icon"
                height="16px"
                viewBox="0 0 120 120"
                width="16px"
              >
                <path
                  d="M69.490332,3.34314575 C72.6145263,0.218951416 77.6798462,0.218951416 80.8040405,3.34314575 C83.8617626,6.40086786 83.9268205,11.3179931 80.9992143,14.4548388 L80.8040405,14.6568542 L35.461,60 L80.8040405,105.343146 C83.8617626,108.400868 83.9268205,113.317993 80.9992143,116.454839 L80.8040405,116.656854 C77.7463184,119.714576 72.8291931,119.779634 69.6923475,116.852028 L69.490332,116.656854 L18.490332,65.6568542 C15.4326099,62.5991321 15.367552,57.6820069 18.2951583,54.5451612 L18.490332,54.3431458 L69.490332,3.34314575 Z"
                  fill="currentColor"
                  fill-rule="nonzero"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next month"
              class="rdp-button_reset rdp-button rdp-nav_button rdp-nav_button_next"
              name="next-month"
              type="button"
            >
              <svg
                class="rdp-nav_icon"
                height="16px"
                viewBox="0 0 120 120"
                width="16px"
              >
                <path
                  d="M49.8040405,3.34314575 C46.6798462,0.218951416 41.6145263,0.218951416 38.490332,3.34314575 C35.4326099,6.40086786 35.367552,11.3179931 38.2951583,14.4548388 L38.490332,14.6568542 L83.8333725,60 L38.490332,105.343146 C35.4326099,108.400868 35.367552,113.317993 38.2951583,116.454839 L38.490332,116.656854 C41.5480541,119.714576 46.4651794,119.779634 49.602025,116.852028 L49.8040405,116.656854 L100.804041,65.6568542 C103.861763,62.5991321 103.926821,57.6820069 100.999214,54.5451612 L100.804041,54.3431458 L49.8040405,3.34314575 Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <table
          aria-labelledby="test-0"
          class="rdp-table"
          id="test-grid-0"
          role="grid"
        >
          <thead
            class="rdp-head"
          >
            <tr
              class="rdp-head_row"
            >
              <th
                aria-label="Sunday"
                class="rdp-head_cell"
                scope="col"
              >
                Su
              </th>
              <th
                aria-label="Monday"
                class="rdp-head_cell"
                scope="col"
              >
                Mo
              </th>
              <th
                aria-label="Tuesday"
                class="rdp-head_cell"
                scope="col"
              >
                Tu
              </th>
              <th
                aria-label="Wednesday"
                class="rdp-head_cell"
                scope="col"
              >
                We
              </th>
              <th
                aria-label="Thursday"
                class="rdp-head_cell"
                scope="col"
              >
                Th
              </th>
              <th
                aria-label="Friday"
                class="rdp-head_cell"
                scope="col"
              >
                Fr
              </th>
              <th
                aria-label="Saturday"
                class="rdp-head_cell"
                scope="col"
              >
                Sa
              </th>
            </tr>
          </thead>
          <tbody
            class="rdp-tbody"
          >
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  1
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  2
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  3
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  4
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  5
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  6
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  7
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  8
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  9
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  10
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  11
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  12
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  13
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  14
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  15
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  16
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  aria-selected="true"
                  class="rdp-button_reset rdp-button rdp-day rdp-day_selected rdp-day_range_end rdp-day_range_start"
                  name="day"
                  role="gridcell"
                  tabindex="0"
                  type="button"
                >
                  17
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  18
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  19
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  20
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  21
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  22
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  23
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  24
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  25
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  26
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  27
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  28
                </button>
              </td>
            </tr>
            <tr
              class="rdp-row"
            >
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  29
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <button
                  class="rdp-button_reset rdp-button rdp-day"
                  name="day"
                  role="gridcell"
                  tabindex="-1"
                  type="button"
                >
                  30
                </button>
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
              <td
                class="rdp-cell"
                role="presentation"
              >
                <div
                  role="gridcell"
                />
              </td>
            </tr>
          </tbody>
          <tfoot
            class="rdp-tfoot"
          >
            <tr>
              <td
                colspan="8"
              >
                <p>
                  November 17th, 2020
                  –
                  November 17th, 2020
                </p>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;
